report/4.Design/Usb.tex

```diff
@@ -0,0 +1,29 @@
+\section{USB communication}
+\label{sec:usbdes}
+As described in the component diagram in Section~\ref{sec:architecture}, the system will consist of two separate hardware platforms, that need to communicate.
+As the NXT can only communicate with external devices using Bluetooth and USB, and we deemed Bluetooth too slow for our system, the group has decided to use USB.
+According to the nxtOSEK documentation, the nxtOSEK library should also support built-in USB, through an easy to use API\cite{ecrobotUSB}.
+When the group tested this functionality, it turned out that the nxtOSEK library for the computer, was no longer functional.
+
+The group tried a few different solutions for custom NXT USB communication, but in the end, writing our own module to handle the communication was deemed the best solution.
+
+The communication is described on Figure~\ref{fig:compusb}:
+
+\figur{1}{images/USBCommunicationDiagram}{Diagram of the USB communication}{fig:compusb}
+
+The USB communication module will consist of three parts.
+
+The initialization part, which will perform a handshake with the NXT, initiating the USB receiving.
+This part posed a bit of a challenge, as the handshake and its content was not documented, thus leaving the group in the dark about its existence.
+The group used a USB sniffer, attached to a working compiled program, a usbtest sample, which is present in the samples directory when downloading the nxtOSEK library\cite{ecrobotUSB}.
```

Comment:
```suggestion
The group used a USB sniffer, attached to a working compiled program, a usb test sample, which is present in the samples directory when downloading the nxtOSEK library\cite{ecrobotUSB}.
```

also maybe something about where the usb test is from (nxtosek examples I assume?)